Sources/MySQLNIO/Commands/MySQLQueryCommand.swift

```diff
@@ -161,6 +162,10 @@ private final class MySQLQueryCommand: MySQLCommand {
             self.state = .executeColumns(remaining: numericCast(count))
             return .noResponse
         case .executeColumns(var remaining):
+            if self.ok!.numColumns == 0 && remaining > 0 {
```

Comment:
From my reading of the code, where does `ok` get set? The force unwrap seems odd with no checks above

Reply:
This is set earlier in the state machine, if it hasn't been set by the time this state is reached, the state machine itself is incorrect (i.e. it's legitimately programmer error; that code path can't be induced by a misbehaving server). The whole state machine is a mess, though, and cleaning it up further is on the remaining todo list for this PR.